pkg/controller/restore_controller.go

```diff
@@ -462,43 +468,118 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 	return nil
 }
 
-func (c *restoreController) fetchBackup(objectStore cloudprovider.ObjectStore, name string) (*api.Backup, error) {
-	backup, err := c.backupLister.Backups(c.namespace).Get(name)
-	if err == nil {
-		return backup, nil
+// fetchBackupInfo checks the backup lister for a backup that matches the given name. If it doesn't
+// find it, it tries to retrieve it from one of the backup storage locations.
+func (c *restoreController) fetchBackupInfo(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	var info backupInfo
+	var err error
+	info.backup, err = c.backupLister.Backups(c.namespace).Get(backupName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return backupInfo{}, errors.WithStack(err)
+		}
+
+		logContext := c.logger.WithField("backupName", backupName)
+		logContext.Debug("Backup not found in backupLister, checking each backup location directly, starting with default...")
+		return c.fetchFromBackupStorage(backupName, pluginManager)
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return nil, errors.WithStack(err)
+	location, err := c.backupLocationLister.BackupStorageLocations(c.namespace).Get(info.backup.Spec.StorageLocation)
+	if err != nil {
+		return backupInfo{}, errors.WithStack(err)
 	}
 
-	logContext := c.logger.WithField("backupName", name)
+	info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+	if err != nil {
+		return backupInfo{}, errors.Wrap(err, "error initializing object store")
+	}
+	info.bucketName = location.Spec.ObjectStorage.Bucket
 
-	logContext.Debug("Backup not found in backupLister, checking object storage directly")
-	backup, err = c.getBackup(objectStore, c.bucket, name)
+	return info, nil
+}
+
+// fetchFromBackupStorage checks each backup storage location, starting with the default,
+// looking for a backup that matches the given backup name.
+func (c *restoreController) fetchFromBackupStorage(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	locations, err := c.backupLocationLister.BackupStorageLocations(c.namespace).List(labels.Everything())
 	if err != nil {
-		return nil, err
+		return backupInfo{}, err
+	}
+
+	orderedLocations := orderedBackupLocations(locations, c.defaultBackupLocation)
+
+	logContext := c.logger.WithField("backupName", backupName)
+	var info backupInfo
+	for _, location := range orderedLocations {
+		info, err = c.backupInfoForLocation(location, backupName, pluginManager)
```

Comment:
how about:
```
info, err := c.backupInfoForLocation(location, backupName, pluginManager)
if err != nil {
    logContext.WithField("locationName", location.Name).WithError(err).Error("Unable to fetch backup from object storage location")
    continue
}
return info, nil
```

Then below the loop, you can remove everything but L523 since we'll only get there if we didn't find the backup anywhere

Reply:
But given that we're now returning from within the loop if we successfully get a result from `backupInfoForLocation`, then if we exit the loop, we know that we failed.  So we should just be able to `return backupInfo{}, errors.New("Unable to get backup from any backup storage location")`, right?

Reply:
Ah yes, I missunderstood your suggestion.  I also personally will do anything to avoid the last return being a negative path statement but yes, this works indeed!